### PR TITLE
Use enums in interfaces

### PR DESCRIPTION
**Pull request by jacobmerson**

### CODE REVIEW

**Comment by jacobmerson** [comment]
@cwsmith can you take a look at this when you get a chance?

File: redev.h [hunk 1]
```diff
@@ -330,24 +338,25 @@ class Redev {
      * Create a Redev object
      * @param[in] comm MPI communicator containing the ranks that are part of the server/client
      * @param[in] ptn Partition object defining the redezvous domain partition
-     * @param[in] isRendezvous true for the server processes and false otherwise
+     * @param[in] processProcess enum for if the server is a client, server, or both
```
Comment (cwsmith) [inline redev.h:341]: I think we should remove the `both` text until we support that

File: redev.h [hunk 2]
```diff
@@ -366,34 +375,45 @@ class Redev {
...
+  std::string engineType;
+  switch (transportType) {
+    case TransportType::BP4 :
+      engineType = "BP4";
+      break;
+    case TransportType::SST:
+      engineType = "SST";
+      break;
+    // no default case. This will cause a compiler error if we do not handle a
+    // an engine type that has been defined in the TransportType enum. (-Werror=switch)
```
Comment (cwsmith) [inline redev.h:397]: Cool!

**Review by cwsmith — APPROVED** [review]
LGTM.  Thank you.

**Comment by jacobmerson** [comment]
@cwsmith this should be good to go.

**Pull request merged**